openff/toolkit/topology/molecule.py

```diff
@@ -298,7 +300,7 @@ def to_dict(self) -> Dict[str, Union[str, int, bool, Dict[Any, Any]]]:
         }
 
     @classmethod
-    def from_dict(cls, atom_dict):
+    def from_dict(cls: Type[A], atom_dict: dict) -> A:
```

Comment:
RE comment on l166: Like here - should we consistently use `Type` or `type` instead of mixing them?

Reply:
Probably so - I don't think I looked at the difference here, owing to some bad experience mixing them on a previous Python version long ago.

Reply:
Memorialized in #1790